homeassistant/components/sensibo/switch.py

```diff
@@ -143,37 +120,77 @@ def is_on(self) -> bool | None:
 
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the entity on."""
-        params = build_params(self.entity_description.command_on, self.device_data)
-        result = await self.async_send_command(
-            self.entity_description.command_on, params
-        )
-
-        if result["status"] == "success":
-            setattr(self.device_data, self.entity_description.remote_key, True)
-            self.async_write_ha_state()
-            return await self.coordinator.async_request_refresh()
-        raise HomeAssistantError(
-            f"Could not execute {self.entity_description.command_on} for device {self.name}"
-        )
+        if self.entity_description.key == "timer_on_switch":
+            await self.turn_on_timer(
+                device_data=self.device_data,
+                key=self.entity_description.data_key,
+                value=True,
+            )
+        if self.entity_description.key == "pure_boost_switch":
+            await self.turn_on_off_pure_boost(
+                device_data=self.device_data,
+                key=self.entity_description.data_key,
+                value=True,
+            )
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the entity off."""
-        params = build_params(self.entity_description.command_on, self.device_data)
-        result = await self.async_send_command(
-            self.entity_description.command_off, params
-        )
-
-        if result["status"] == "success":
-            setattr(self.device_data, self.entity_description.remote_key, False)
-            self.async_write_ha_state()
-            return await self.coordinator.async_request_refresh()
-        raise HomeAssistantError(
-            f"Could not execute {self.entity_description.command_off} for device {self.name}"
-        )
+        if self.entity_description.key == "timer_on_switch":
+            await self.turn_off_timer(
+                device_data=self.device_data,
+                key=self.entity_description.data_key,
+                value=True,
+            )
+        if self.entity_description.key == "pure_boost_switch":
+            await self.turn_on_off_pure_boost(
+                device_data=self.device_data,
+                key=self.entity_description.data_key,
+                value=True,
+            )
 
     @property
     def extra_state_attributes(self) -> Mapping[str, Any] | None:
         """Return additional attributes."""
         if self.entity_description.extra_fn:
             return self.entity_description.extra_fn(self.device_data)
         return None
+
+    @api_call_decorator
+    async def turn_on_timer(
```

Comment:
```suggestion
    async def async_turn_on_timer(
```

Since Home Assistant has both async and non-async functions, we usually prefix async safe one with `async_`